### Eclipse Workspace Patch 1.0
#P org.eclipse.linuxtools.rpm.ui.editor
Index: src/org/eclipse/linuxtools/rpm/ui/editor/RpmPackageProposalsList.java
===================================================================
--- src/org/eclipse/linuxtools/rpm/ui/editor/RpmPackageProposalsList.java	(revision 24089)
+++ src/org/eclipse/linuxtools/rpm/ui/editor/RpmPackageProposalsList.java	(working copy)
@@ -52,6 +52,8 @@
 					list.add(line.trim());
 					line = reader.readLine();
 				}
+			} else {
+				RpmPackageBuildProposalsJob.update();
 			}
 		} catch (IOException e) {
 			RpmPackageBuildProposalsJob.update();
Index: src/org/eclipse/linuxtools/rpm/ui/editor/preferences/RpmProposalsPreferencePage.java
===================================================================
--- src/org/eclipse/linuxtools/rpm/ui/editor/preferences/RpmProposalsPreferencePage.java	(revision 24089)
+++ src/org/eclipse/linuxtools/rpm/ui/editor/preferences/RpmProposalsPreferencePage.java	(working copy)
@@ -11,6 +11,7 @@
 
 package org.eclipse.linuxtools.rpm.ui.editor.preferences;
 
+import java.io.File;
 import java.util.ArrayList;
 
 import org.eclipse.jface.preference.BooleanFieldEditor;
@@ -18,6 +19,7 @@
 import org.eclipse.jface.preference.FieldEditorPreferencePage;
 import org.eclipse.jface.preference.RadioGroupFieldEditor;
 import org.eclipse.jface.preference.StringFieldEditor;
+import org.eclipse.jface.util.PropertyChangeEvent;
 import org.eclipse.linuxtools.rpm.core.utils.Utils;
 import org.eclipse.linuxtools.rpm.ui.editor.Activator;
 import org.eclipse.swt.SWT;
@@ -37,6 +39,8 @@
 public class RpmProposalsPreferencePage extends FieldEditorPreferencePage
 		implements IWorkbenchPreferencePage {
 
+	private FieldEditor rpmtools;
+
 	/*
 	 * default constructor
 	 */
@@ -52,7 +56,8 @@
 	 */
 	@Override
 	public void createFieldEditors() {
-		addField(rpmtoolsRadioGroupFieldEditor());	
+		rpmtools = rpmtoolsRadioGroupFieldEditor();
+		addField(rpmtools);	
 		// FIXME: there is validations problem when a FileFieldEditor is used, so 
 		// as a quick fix, StringFieldEditor is used.
 		StringFieldEditor rpmListFieldEditor = new StringFieldEditor(PreferenceConstants.P_RPM_LIST_FILEPATH,
@@ -132,4 +137,23 @@
 
 	}
 
+	/* (non-Javadoc)
+	 * @see org.eclipse.jface.preference.FieldEditorPreferencePage#propertyChange(org.eclipse.jface.util.PropertyChangeEvent)
+	 */
+	@Override
+	public void propertyChange(PropertyChangeEvent event) {
+		super.propertyChange(event);
+		if (event.getSource().equals(rpmtools)) {
+			if (!event.getOldValue().equals(event.getNewValue())) {
+				String rpmpkgsFile = Activator.getDefault()
+						.getPreferenceStore().getString(
+								PreferenceConstants.P_RPM_LIST_FILEPATH);
+				if (Utils.fileExist(rpmpkgsFile)) {
+					new File(rpmpkgsFile).delete();
+				}
+				Activator.packagesList = null;
+			}
+		}
+	}
+
 }
